presto-parquet/src/main/java/com/facebook/presto/parquet/reader/ParquetReader.java

```diff
@@ -132,7 +135,8 @@ public ParquetReader(MessageColumnIO
             boolean enableVerification,
             Predicate parquetPredicate,
             List<ColumnIndexStore> blockIndexStores,
-            boolean columnIndexFilterEnabled)
+            boolean columnIndexFilterEnabled,
+            InternalFileDecryptor fileDecryptor)
```

Comment:
I might suggest to use Optional< InternalFileDecryptor > to avoid null values

Reply:
sure